modules/test/playwright/pages/commerce/commerce-product-definitions-web/attachmentsPage.ts

```diff
@@ -0,0 +1,149 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Locator, Page} from '@playwright/test';
+
+import {CommerceLayoutsPage} from '../commerceLayoutsPage';
+
+export class AttachmentsPage {
+	readonly addFileEntryButton: Locator;
+	readonly addFileEntrySubmitButton: Locator;
+	readonly contentAndDataMenuItem: Locator;
+	readonly documentsAndMediaMenuItem: Locator;
+	readonly externalVideoShortcutMenuItem: Locator;
+	readonly externalVideoShortcutURLInput: Locator;
+	readonly fileUploadMenuItem: Locator;
+	readonly layoutsPage: CommerceLayoutsPage;
+	readonly page: Page;
+	readonly renderedVideoPreviewIframe: Locator;
+	readonly selectFileButton: Locator;
+	readonly titleInput: Locator;
+	readonly uploadedFileName: Locator;
+
+	constructor(page: Page) {
+		this.addFileEntryButton = page.getByTestId('creationMenuNewButton');
+		this.addFileEntrySubmitButton = page
+			.getByTestId('addFileEntryFooter')
+			.getByRole('button', {exact: true, name: 'Publish'});
+		this.contentAndDataMenuItem = page
+			.getByTestId('appGroup')
+			.filter({hasText: 'Content & Data'});
+		this.documentsAndMediaMenuItem = page
+			.getByTestId('app')
+			.filter({hasText: 'Documents and Media'});
+		this.externalVideoShortcutMenuItem = page
+			.getByTestId('dropdownMenu')
+			.getByRole('menuitem', {
+				exact: true,
+				name: 'External Video Shortcut',
+			});
+		this.externalVideoShortcutURLInput = page.locator(
+			'#dlVideoExternalShortcutURLInput'
+		);
+		this.fileUploadMenuItem = page
+			.getByTestId('dropdownMenu')
+			.getByRole('menuitem', {
+				exact: true,
+				name: 'File Upload',
+			});
+		this.layoutsPage = new CommerceLayoutsPage(page);
+		this.page = page;
+		this.renderedVideoPreviewIframe = page.getByTestId('renderedVideo');
+		this.selectFileButton = page.getByTestId('selectFileButton');
+		this.titleInput = page.locator(
+			'#_com_liferay_document_library_web_portlet_DLAdminPortlet_title'
+		);
+		this.uploadedFileName = page.getByTestId('uploadedFileName');
+	}
+
+	async createExternalVideoShortcut(url: string, title: string) {
+		await this.addFileEntryButton.click();
+		await this.externalVideoShortcutMenuItem.click();
+		await this.externalVideoShortcutURLInput.waitFor({
+			state: 'attached',
+		});
+		await this.externalVideoShortcutURLInput.click();
```

Comment:
Just one question because I don't know... do you have to click on the input field before fill it? Or the fill(string) works also without the previous click?

Reply:
@smottal , we need to click it first. I have noticed failures related to the fill(string) bugging out too many times which is only fully solved by clicking the element first.